src/plugins/security_oss/server/plugin.ts

```diff
@@ -17,23 +17,64 @@
  * under the License.
  */
 
-import { CoreSetup, Logger, Plugin, PluginInitializerContext } from 'kibana/server';
+import {
+  Capabilities,
+  CoreSetup,
+  KibanaRequest,
+  Logger,
+  Plugin,
+  PluginInitializerContext,
+} from 'kibana/server';
 import { BehaviorSubject, Observable } from 'rxjs';
 import { createClusterDataCheck } from './check_cluster_data';
 import { ConfigType } from './config';
-import { setupDisplayInsecureClusterAlertRoute } from './routes';
+import { setupAppStateRoute, setupAnonymousAccessCapabilitiesRoute } from './routes';
 
 export interface SecurityOssPluginSetup {
   /**
    * Allows consumers to show/hide the insecure cluster warning.
    */
   showInsecureClusterWarning$: BehaviorSubject<boolean>;
+
+  /**
+   * Set the provider function that returns a service that can deal with various aspects of the
+   * anonymous access.
+   * @param provider
+   */
+  setAnonymousAccessServiceProvider: (provider: () => AnonymousAccessService) => void;
+}
+
+export interface AnonymousAccessService {
+  /**
+   * Indicates whether anonymous access is enabled.
+   */
+  readonly isAnonymousAccessEnabled: boolean;
+
+  /**
+   * A map of query string parameters that should be specified in URL so that anonymous user can use
+   * to automatically log in to Kibana.
+   */
+  readonly accessURLParameters: Readonly<Map<string, string>> | null;
+
+  /**
+   * Gets capabilities of the anonymous service account.
+   * @param request Kibana request instance.
+   */
+  getCapabilities: (request: KibanaRequest) => Promise<Capabilities>;
 }
 
 export class SecurityOssPlugin implements Plugin<SecurityOssPluginSetup, void, {}, {}> {
   private readonly config$: Observable<ConfigType>;
   private readonly logger: Logger;
 
+  private anonymousAccessServiceProvider?: () => AnonymousAccessService;
```

Comment:
__note:__ kind of clumsy, but I cannot think of a better approach to pass service that is only available at `start`.

Reply:
Agreed. It's at least consistent with the way we've done this elsewhere 🤷‍♂️